### ISSUE TITLE
Wifislax 2.4 hostbase-1.1 mdk3 deauths multiple networks  And doesn't support channel hopping 

### ISSUE THREAD

**Issue by minanagehsalalma** [Koala633/hostbase#9]
Testing the wifislax version on latest wifislax 
The Fake AP shows up without problems 

Steps 
Followed from the YouTube video [Hostbase 2021 actualizacion](https://www.youtube.com/watch?v=c-3oi_vxXbQ)
1. copied `hostbase-1.1Wifislax`  to `/tmp/ `
2. Renamed it to `hostbase-1.1`
3. Copied `paginasAQUI` to `/etc/`
4. copied `httpd.conf` to `/etc/httpd/`
5. cd hostbase-1.1 
6. ruby hostbase.rb
7.  Run The scan
8. Hostapd 1 WPA AP then clicked on button  `Hostapd con 1 red en WPA para mas discrecion

```
wifislax64 hostbase-1.1 # ruby hostbase.rb
3
[*] Ap channel is OK.
Te:3E:B3:40:7B:Te
[*] Ap mac is OK.
wlan0
[*] Wifi card is OK.
orangewps
[*] Phishing page is OK.
4
[*] Channel is OK.
testtttttttt
Falso AP en wlan0, cambiamos el falso AP en wlan0 para no tener conflicto y errores despuès...
[*] Configuracion de hostapd...
Internet Systems Consortium DHCP Server 4.4.1
Copyright 2004-2018 Internet Systems Consortium.
All rights reserved.
For info, please visit https://www.isc.org/software/dhcp/
Config file: /tmp/hostbase-1.1/dhcpd.conf
Database file: /tmp/hostbase-1.1/dhcpd.leases
PID file: /var/run/dhcpd.pid
There's already a DHCP server running.

If you think you have received this message due to a bug rather
than a configuration issue please read the section on submitting
bugs on either our web page at www.isc.org or in the README file
before submitting a bug.  These pages explain the proper
process and the information we find helpful for debugging.

exiting.
dnsspoof: listening on wlan0 [udp dst port 53 and not src 175.0.0.1]
orangewps
[*] Configuracion del systemo de hotspot...
chown: invalid group: ‘root:www-data’
[*] Rogue AP creado... preparacion de la DoS activa que sigue el AP...
wlan0
OK
wlan5
Starting your second wifi-card...
Active DoS will start now...
Esperamos el usario... apoya ctrl+c si quieres salir...                                                                                                                                     
[*] Usario conectado.                                                                                                                                                                       
xterm: no process found
Iniciando wpa_cli con la segunda tarjeta...
wlan5
Te:3E:B3:40:7B:Te
orangewps
Esperamos por el WPS push boton...
wpa_supplicant: no process found

```

Tested if it connects if  i press the button too 
and it worked

```
[*] El boton WPS ha sido apoyado, WPA-PSK:
ctrl_interface=/var/run/wpa_supplicant
ctrl_interface_group=0
update_config=1

network={
        ssid="aaaaaaaaaaaaaaaaaaaaaaaaa"
        psk="dasdsadasdsad"
        proto=WPA
        key_mgmt=WPA-PSK
        pairwise=CCMP GCMP
        group=CCMP GCMP TKIP
        auth_alg=OPEN
        pbss=2
}


```

when closing the attack 
```
^C[*] Salida del programo... limpiando los ficheros... reiniciando las tarjetas por defecto... ESPERA !
Limpieza de ficheros terminado... reiniciando las tarjetas por defecto y network-manager...
sleep: no process found
[+] Starting NetworkManager daemon:  /usr/sbin/NetworkManager 
Bye...
/tmp/hostbase-1.1/dosbis.rb:39:in `exit': exit
         from /tmp/hostbase-1.1/dosbis.rb:39:in `block in <top (required)>'
         from /tmp/hostbase-1.1/dos.rb:125:in `sleep'
         from /tmp/hostbase-1.1/dos.rb:125:in `wpaKey'
         from /tmp/hostbase-1.1/dos.rb:116:in `block (2 levels) in victimeAttente'
         from /tmp/hostbase-1.1/dos.rb:88:in `each'
         from /tmp/hostbase-1.1/dos.rb:88:in `block in victimeAttente'
         from /tmp/hostbase-1.1/dos.rb:77:in `loop'
         from /tmp/hostbase-1.1/dos.rb:77:in `victimeAttente'
         from /tmp/hostbase-1.1/dos.rb:70:in `activeDos'
         from /tmp/hostbase-1.1/dos.rb:64:in `interface'
         from /tmp/hostbase-1.1/rogue.rb:290:in `block in hostapdInit'
         from /tmp/hostbase-1.1/rogue.rb:202:in `each'
         from /tmp/hostbase-1.1/rogue.rb:202:in `hostapdInit'
         from /tmp/hostbase-1.1/rogue.rb:172:in `hostapdNormal'
         from hostbase.rb:116:in `block (2 levels) in <main>'
         from hostbase.rb:107:in `run'
         from hostbase.rb:107:in `block in <main>'
         from hostbase.rb:129:in `main'
         from hostbase.rb:129:in `<main>'

```
What's the problems
1. mdk3 deauths other networks too And doesn't support channel hopping which means it doesn't follow the AP channel when changed 
![Capture11](https://user-images.githubusercontent.com/20546638/150600370-1ec7d959-f06e-4e3a-a029-31c411e230a9.JPG)

2. Why there is no option to make the Fake AP open just like regular evil twin but with WPS PBC phishing , Cause not ALL networks has a windows PC connected to them!! Most if not ALL has mobile devices onlyyyyyy

For that i tried using the older legacy versions but they doesn't work at all !
AP doesn't connect or doesn't launch and mdk3 says unknown flag --g 
and it doesn't connect when the WPS button is pushed 

3. Why the scan APs option doesn't work the same as fluxion 
it scans and lets you choose the AP by just typing a number 

4. How do i customize the WPS PBC phishing Page ? 
I just want a page showing the back of the router and asking users to push it , nothing the less 

5. the deauth needs improvement, Can't it only Stop when the button is pushed and nothing else ? like monitor the AP for wps status and when you notice that it's pushed then stop mdk3 and connect to it 

6. The Whole attack should stop automatically when the password is caputred 

7. The files copying should be automated !! why you leave it to the users to copy them ?

Finally : IF you are willing to fix any of these Issues 
I volunteer to translate the tool to English and Add phishing pages for two countries or something 

Regards 

### A Great Improvement would be a way to detect if the button is pressed or not without trying to connect to it 
that way we wouldn't need to turn off the mdk3 deauth 
or even need to try to connect every couple of seconds 

maybe using a method like this [ wifiphisher wpspbc extension ](https://github.com/wifiphisher/wifiphisher/blob/master/wifiphisher/extensions/wpspbc.py)

### **@kcdtv  Check this I found a simple script that does it** 

but it works on older systems with python 2.7 only 
[wps2key.py](https://github.com/Tourountzis/wifi-pwn/blob/master/wps2key.py)

![Share](https://user-images.githubusercontent.com/20546638/150663553-e042b5b2-ebc9-4cfc-822e-2b3b3c100bfa.png)

It can also be done using wash -i mon0 -j 
![image](https://user-images.githubusercontent.com/20546638/150699904-96c412c6-c408-4a20-a4b3-6f985604f17d.png)

**Comment by Koala633** [comment]
Hi,

First sorry for the delay of my answer.

The hostbase version for wifislax is outdated, i only keep it on github to have a back up and work with it if i make change in the futur.

```
What's the problems

1 mdk3 deauths other networks too And doesn't support channel hopping which means it doesn't follow the AP channel when changed

2  Why there is no option to make the Fake AP open just like regular evil twin but with WPS PBC phishing , Cause not ALL networks has a windows PC connected to them!! Most if not ALL has mobile devices onlyyyyyy

For that i tried using the older legacy versions but they doesn't work at all !
AP doesn't connect or doesn't launch and mdk3 says unknown flag --g
and it doesn't connect when the WPS button is pushed

3 Why the scan APs option doesn't work the same as fluxion
it scans and lets you choose the AP by just typing a number

4 How do i customize the WPS PBC phishing Page ?
I just want a page showing the back of the router and asking users to push it , nothing the less

5 the deauth needs improvement, Can't it only Stop when the button is pushed and nothing else ? like monitor the AP for wps status and when you notice that it's pushed then stop mdk3 and connect to it

6 The Whole attack should stop automatically when the password is caputred

7 The files copying should be automated !! why you leave it to the users to copy them ?

Finally : IF you are willing to fix any of these Issues
I volunteer to translate the tool to English and Add phishing pages for two countries or something

Regards

A Great Improvement would be a way to detect if the button is pressed or not without trying to connect to it
that way we wouldn't need to turn off the mdk3 deauth
or even need to try to connect every couple of seconds

```

For 1: This issue has been fixed using wash instead of airodump-ng.For some reasons airodump-ng can do a false positive when scanning the network channel even if the bssid and channel is specified

For 2: It will be included in the next version with a potential mac filter because we need to know the client wich connect to our fake AP is a legitimate client of the real AP and not a unknown user.

For 3: I don't know how do that but i have to think about it

For 4: The wps_pbc pages are different between each internet service provider (ISP) In the hostbase of wifislax like in the hostbaseES for kali-linux, the phishing page is not the same.I leave in France and i have prepared others phishing page for France to adapt them based on the ISP.In the last version of hostbase i puted a "404" page with a screen of google without internet and a input area on the right side to enter the key.I use it when i don't know wich router is.To modify the phishing page it's easy, you could create it with a web server (in my case apache2 for linux) and then take a picture from the back side of your router and put the picture on your html page.But carrefull each ISP use different type of routers.Once you have modificated the web page, you can go to check.rb and add your new phishing page to the filter.


For 5:  Mmmmh... that depend which router you are making the attack.Some routers take time to wake up after a deauth attack and others come back quicly as their originals states after the attack.For the moment the deauth is stoping when an client connect to the fake AP.Other thing might be a problem, if we stop the deauth when the wps button is pushed, if the victim go to his network area and click to his own network, it will maybe join the network first our attacking machine and the wps_pbc can allow only one device in 120 seconds.I have to test this case.

For 6: It will be fine, yes also i have to include an option if the real AP shutdown to stop the fake AP

for 7: Yeah, i will include that


```
A Great Improvement would be a way to detect if the button is pressed or not without trying to connect to it
that way we wouldn't need to turn off the mdk3 deauth
or even need to try to connect every couple of seconds

```

We need stop stop the mdk3 deauth to use the wps_pbc of the router as quickly as we can because like i said before, if the client connect to his network first we are out of the game and one wps_pbc session is juste 120 seconds.Let's try this, i will include this action if it work fine.


Finaly, thanks you for your interess and your comments :) it is like that the things is going better !

**Comment by minanagehsalalma** [comment]
> First sorry for the delay of my answer.

@Koala633  No problem : ) 

Sorry for my late reply 

> For 1: This issue has been fixed using wash instead of airodump-ng.For some reasons airodump-ng can do a false positive when scanning the network channel even if the bssid and channel is specified

what logic did you follow for using wash for channel tracking ? 

> For 2: It will be included in the next version with a potential mac filter because we need to know the client wich connect to our fake AP is a legitimate client of the real AP and not a unknown user.

I think that wouldn't help much because of mac address randomization is enabled by default for most devices nowadays 

> We need stop stop the mdk3 deauth to use the wps_pbc of the router as quickly as we can because like i said before, if the client connect to his network first we are out of the game and one wps_pbc session is juste 120 seconds.Let's try this, i will include this action if it work fine.
> 
> Finaly, thanks you for your interess and your comments :) it is like that the things is going better !
> 
> 

Thanks a lot you are welcome : )

**Comment by Koala633** [comment]
Hi :)

`what logic did you follow for using wash for channel tracking ?`

I use wash with the channel option and after the command is launched if he detect the bssid that mean the AP is on the last channel tested.

` think that wouldn't help much because of mac address randomization is enabled by default for most devices nowadays`


Yeah i thinked more about that and it is a bad idea, i don't use this on the last version.

I just released the 1.4 version for kali-linux but it is totally expérimental and since yesterday when i updated my computer to kali 2022.1 some options like the wash loop is not working fine.I don't know what happen maybe a light diference between the ruby version, i have to work on it to fix it

**Comment by minanagehsalalma** [comment]
> I use wash with the channel option and after the command is launched if he detect the bssid that mean the AP is on the last channel tested.

@Koala633  hmm if so , then why can't the attack be done using one adapter only ? 


> I just released the 1.4 version for kali-linux but it is totally expérimental and since yesterday when i updated my computer to kali 2022.1 some options like the wash loop is not working fine.I don't know what happen maybe a light diference between the ruby version, i have to work on it to fix it

oh what code did you use for the loop ? 

also i think it has something to do with revear since it has to be the latest version 

update revear then try again : )
